src/btree/bt_vrfy.c

```diff
@@ -1347,6 +1066,36 @@ __verify_page_cell(
               session, NULL, ref, cell_num - 1, unpack.start_ts, unpack.stop_ts, vs));
             break;
         }
+
+        /*
+         * Verify key-associated history-store entries, optionally dump historical time pairs and
+         * values in debug mode.
+         */
+        if (page->type == WT_PAGE_ROW_LEAF) {
+            if (unpack.type != WT_CELL_KEY && unpack.type != WT_CELL_KEY_OVFL)
+                continue;
+
+            WT_RET(__wt_row_leaf_key(session, page, rip++, vs->tmp1, false));
+            WT_RET(__verify_key_hs(session, vs->tmp1, unpack.start_ts, vs));
+
+#ifdef HAVE_DIAGNOSTIC
+            if (vs->dump_history)
+                WT_RET(__wt_debug_key_value(session, vs->tmp1, WT_RECNO_OOB, 0, &unpack));
+#endif
+        } else if (page->type == WT_PAGE_COL_VAR) {
+            rle = __wt_cell_rle(&unpack);
+            p = vs->tmp1->mem;
+            WT_RET(__wt_vpack_uint(&p, 0, recno));
+            vs->tmp1->size = WT_PTRDIFF(p, vs->tmp1->mem);
+            WT_RET(__verify_key_hs(session, vs->tmp1, unpack.start_ts, vs));
+
+#ifdef HAVE_DIAGNOSTIC
```

Comment:
Is it possible to maintain consistency and have the __wt_debug_key_value call prior to the __verify_key_hs call?